cfgmgr/tunnelmgr.cpp

```diff
@@ -223,6 +224,7 @@ bool TunnelMgr::doTunnelTask(const KeyOpFieldsValuesTuple & t)
 
     const std::string & tunnelName = kfvKey(t);
     const std::string & op = kfvOp(t);
+    std::unique_ptr<std::string> src_ip;
```

Comment:
It's smart_ptr required here? I prefer using string if not necessary

Reply:
Because we are checking the `src_ip` presence later in this function, so keeping `src_ip` as a pointer could differentiate the difference between `absence` and `presence`. Using string has one edge issue that if the `src_ip` stored in redis is an empty string.

Reply:
updated.